packages/flutter_tools/lib/src/ios/xcresult.dart

```diff
@@ -295,3 +315,60 @@ String? _convertUrlToLocationString(String url) {
   }
   return '${fileLocation.path}$startingLineNumber$startingColumnNumber';
 }
+
+// Determine if an `issue` should be discarded based on the `discarder`.
+bool _shouldDiscardIssue(
+    {required XCResultIssue issue, required XCResultIssueDiscarder discarder}) {
+  if (issue.type == discarder.typeMatcher) {
+    return true;
+  }
+  if (issue.subType != null &&
+      discarder.subTypeMatcher != null &&
+      discarder.subTypeMatcher!.hasMatch(issue.subType!)) {
+    return true;
+  }
+  if (issue.message != null &&
+      discarder.messageMatcher != null &&
+      discarder.messageMatcher!.hasMatch(issue.message!)) {
+    return true;
+  }
+  if (issue.location != null &&
+      discarder.locationMatcher != null &&
+      discarder.locationMatcher!.hasMatch(issue.location!)) {
+    return true;
+  }
+
+  return false;
+}
+
+List<XCResultIssue> _parseIssuesFromIssueSummariesJson({
```

Comment:
Move this private function outside of the `generate' function for cleaner code. 
The changes here are adding the discarder logic.